firebase.agent.nut

```diff
@@ -384,51 +384,95 @@ class Firebase {
+            return [];
+        }
+
+        local parsedEvents = [];
+
+        for (local i = 0; i < allLines.len(); ) {
+            local lines = [];
+            //try to get 2 lines
+            if (i + 1 < allLines.len()) {
+                lines.push(allLines[i++]);
+                lines.push(allLines[i++]);
+            } else {
+                // check, if we have at least one line, that we should to save
+                if (i < allLines.len()) {
+                    isEndLine = text[text.len() - 1] == "\n";
+                    _bufferedInput = lines[i] + (isEndLine ? "\n" : "");
+                }
+                return parsedEvents; 
             }
 
-            // Check for error conditions
-            if (lines.len() == 3 && lines[0] == "{" && lines[2] == "}") {
-                local error = http.jsondecode(text);
-                _logError("Firebase error message: " + error.error);
-                continue;   //The continue operator jumps to the next iteration of the loop skipping the execution of the following statements.
+            // Error have 3 lines and last one is "}"
+            if (i < allLines.len() && allLines[i] == "}") {
+                lines.push(allLines[i++]);
+                try {
+                    local error = http.jsondecode(text);
+                    _logError("Firebase error message: " + error.error);
+                } catch (e) { 
+                    _logError("Exeption while parsing error message: " + e);
+                }
+                continue;   // The continue operator jumps to the next iteration of the loop skipping the execution of the following statements.
             }
 
             // get the event
             local eventLine = lines[0];
             local event = eventLine.slice(7);
 
-            if(event.tolower() == "keep-alive") continue;
+            // keep-alive contains no data
+            if (event.tolower() == "keep-alive") continue;
 
             // get the data
             local dataLine = lines[1];
             local dataString = dataLine.slice(6);
 
             // pull interesting bits out of the data
-            local d;
+            local d = null;
             try {
+                // try to encode json to get data and path fields
                 d = http.jsondecode(dataString);
             } catch (e) {
-                _logError("Exception while decoding (" + dataString.len() + " bytes): " + dataString);
-                throw e;
+                // Check, if it is last line and we want to wait another part, or
+                // message is broken
+                if (i + 1 < allLines.len()) {
+                    _logError("Exception while decoding (" + dataString.len() + ") bytes): " + dataString);
+                    _bufferedInput = "";
+                    continue; 
+                } else {
+                    // add last not full message to buffer
+                    local isEndLine = text[text.len() - 1] == "\n";
```

Comment:
I'm confused here. Why is that? Can you please bring an example on why is this check for?